ursabot/docker.py

```diff
@@ -265,87 +277,55 @@ def conda(*packages, files=tuple()):
     return cmd.lstrip()
 
 
-docker = Path(__file__).parent.parent / 'docker'
+# configure shell and entrypoint to load .bashrc
+docker_assets = Path(__file__).parent.parent / 'docker'
 
-# TODO(kszucs): add buildbot user
-worker_command = 'twistd --pidfile= -ny buildbot.tac'
-worker_steps = [
-    RUN(pip('buildbot-worker', executable='pip3')),
-    RUN(mkdir('/buildbot')),
-    ADD(docker / 'buildbot.tac', '/buildbot/buildbot.tac'),
-    WORKDIR('/buildbot'),
-    CMD(worker_command)  # not this is string!
-]
-conda_worker_steps = (
-    [RUN(conda('twisted')),
-     RUN(pip('buildbot-worker', executable='pip'))] +
-    worker_steps[1:] +
-    [CMD([worker_command])]  # note this is list!
-)
+ubuntu_pkgs = (docker_assets / 'pkgs-ubuntu.txt').read_text().splitlines()
+alpine_pkgs = (docker_assets / 'pkgs-alpine.txt').read_text().splitlines()
 python_steps = [
-    ADD(docker / 'requirements.txt'),
-    ADD(docker / 'requirements-test.txt'),
-    RUN(pip('cython', files=['requirements.txt'], executable='pip3')),
-    RUN(pip(files=['requirements-test.txt'], executable='pip3'))
+    ADD(docker_assets / 'requirements.txt'),
+    ADD(docker_assets / 'requirements-test.txt'),
+    RUN(pip('cython', files=['requirements.txt'])),
+    RUN(pip(files=['requirements-test.txt']))
 ]
 
-ursabot_images = ImageCollection([
-    DockerImage('ursabot', base='python:3.7', arch='amd64', os='debian',
-                tag='worker', steps=worker_steps)
-])
-
+# Note the python has a special treatment, because buildbot requires it.
+# So all of the following images must have a python interpreter and pip
+# pre-installed.
 arrow_images = ImageCollection()
 
-ubuntu_pkgs = (docker / 'pkgs-ubuntu.txt').read_text().splitlines()
-alpine_pkgs = (docker / 'pkgs-alpine.txt').read_text().splitlines()
-
 for arch in ['amd64', 'arm64v8']:
     # UBUNTU
     for version in ['16.04', '18.04']:
         os = f'ubuntu-{version}'
         base = f'{arch}/ubuntu:{version}'
 
         cpp = DockerImage('cpp', base=base, arch=arch, os=os, steps=[
-            RUN(apt(*ubuntu_pkgs))
+            RUN(apt(*ubuntu_pkgs, 'python3', 'python3-pip')),
+            RUN(symlink({'/usr/bin/python': '/usr/bin/python3',
```

Comment:
Symlinking in `/usr/bin` is a dangerous game, you can easily break the base OS like this, e.g. `yum` used to expect python2. Does it work if you use `/usr/local/bin/python`?

Reply:
Good point!

Reply:
@fsaintjacques fixed